src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -419,19 +418,10 @@ class SchedulerActions(
       }
     }
 
-    if (instancesToStart.isDefined) {
-      val toStart = instancesToStart.get
-
-      logger.info(s"Need to scale ${runSpec.id} from ${runningInstances.size} up to $targetCount instances")
-      val leftToLaunch = scheduledInstances.size
-      val toAdd = toStart - leftToLaunch
-
-      if (toAdd > 0) {
-        logger.info(s"Queueing $toAdd new instances for ${runSpec.id} to the already $leftToLaunch queued ones")
-        await(launchQueue.add(runSpec, toAdd))
-      } else {
-        logger.info(s"Already queued ${scheduledInstances.size} and started ${runningInstances.size} instances for ${runSpec.id}. Not scaling.")
-      }
+    val toStart = instancesToStart.getOrElse(0)
+    if (toStart > 0) {
```

Comment:
Not sure if async allows this but how about

```scala
instancesToStart.foreach { toStart =>
  await(launchQueue.add(runSpec, toStart))
}
```

Reply:
It doesn't 😞